src/NLog.Targets.Syslog/MessageSend/Tcp.cs

```diff
@@ -66,7 +67,7 @@ private bool IsSocketConnected()
             if (connectionCheckTimeout <= 0)
                 return true;
 
-            return  tcp.Client.Poll(connectionCheckTimeout, SelectMode.SelectRead) && tcp.Client.Available == 0;
+            return  tcp.Client.Poll(connectionCheckTimeout, SelectMode.SelectWrite) && tcp.Client.Available == 0;
```

Comment:
This change is not related with this PR (it should have its own PR).
Anyway SelectRead should be used:
     - https://msdn.microsoft.com/en-us/library/system.net.sockets.socket.poll.aspx
     - http://stackoverflow.com/questions/722240/instantly-detect-client-disconnection-from-server-socket
     - http://blog.stephencleary.com/2009/05/detection-of-half-open-dropped.html